common/lib/util_sys.c

```diff
@@ -83,113 +75,124 @@ __weak int pal_submit_12v_cycle_slot()
 	return NOT_SUPPORT_12V_CYCLE_SLOT;
 }
 
-#define ME_FW_RECOVERY 0x01
-#define ME_FW_RESTORE 0x02
-
-static void set_ME_FW_mode(uint8_t me_fw_mode)
+/* The byte-2 of ME response for "Get Self-Test Result" command */
+enum GET_SELF_TEST_RESULT_RESPONSE {
+	NO_ERROR = 0x55,
+	FIRMWARE_ENTERED_RECOVERY_MODE = 0x81,
+};
+
+enum RECOVERY_MODE_CAUSE {
+	RECOVERY_MODE_CAUSE_RECOVERY_JUMPER = 0,
+	RECOVERY_MODE_CAUSE_SECURITY_STRAP_OVERRIDE_JUMPER,
+	RECOVERY_MODE_CAUSE_IPMI_COMMAND,
+	RECOVERY_MODE_CAUSE_INVALID_FLASH_CONFIG,
+	RECOVERY_MODE_CAUSE_INTERNAL_ERROR
+};
+
+void set_me_firmware_mode(uint8_t me_fw_mode)
 {
-	ipmi_msg *me_msg;
-	ipmb_error status;
-
-	me_msg = (ipmi_msg *)malloc(sizeof(ipmi_msg));
-	if (me_msg == NULL) {
-		printf("[%s] Failed to allocate memory\n", __func__);
+	if ((me_fw_mode != ME_FW_RECOVERY) && (me_fw_mode != ME_FW_RESTORE)) {
+		printf("Not support the ME frimware mode 0x%x setting", me_fw_mode);
```

Comment:
"Unsupported ME firmware mode 0x%x setting"

Reply:
We have fixed this and done the forced update.
Please review it. Thanks.